Tests/FoundationEssentialsTests/AttributedString/AttributedStringTests.swift

```diff
@@ -2599,4 +2599,26 @@ E {
         XCTAssertEqual(testConstrainedContainer.filter(inheritedByAddedText: true), AttributeContainer.testInt(2).testParagraphConstrained(3).testCharacterConstrained(4))
         XCTAssertEqual(testConstrainedContainer.filter(inheritedByAddedText: false), AttributeContainer.testNonExtended(5))
     }
+
+    func testWritingDirectionBehavior() {
+        // Indicate that this sentence is primarily right to left, because the English term "Swift" is embedded into an Arabic sentence.
+        var string = AttributedString("Swift مذهل!", attributes: .init().writingDirection(.rightToLeft))
+
+        // To remove the information about the writing direction, set it to `nil`:
+        string.writingDirection = nil
+
+        let range = string.range(of: "Swift")!
+
+        // When setting or removing the value from a certain range, the value will always be applied to the entire paragraph(s) that intersect with that range:
+        string[range].writingDirection = .leftToRight
+        assert(string.runs[\.writingDirection].count == 1)
```

Comment:
```suggestion
        XCTAssertEqual(string.runs[\.writingDirection].count, 1)
```

In general we prefer `XCTest` assertions here rather than `assert` since `assert` is only evaluated in debug builds and will crash the test runner if there's a bug in the output rather than providing a helpful message / output.